API Keys content update (#3594)
temporalio/documentation: docs/develop/go/temporal-client.mdx

## Connect to Temporal Cloud {#connect-to-temporal-cloud}

### How to connect to Temporal Cloud using an API Key {#connect-to-temporal-cloud-api-key}

To use an [API key](/cloud/api-keys) with the Temporal Go SDK, use the following code example.

Create an initial connection:

```go
clientOptions := client.Options{
    HostPort: <endpoint>,
    Namespace: <namespace_id>.<account_id>,

>     ConnectionOptions: client.ConnectionOptions{TLS: &tls.Config{},},

Reviewer: ```suggestion
    ConnectionOptions: client.ConnectionOptions{TLS: &tls.Config{}},
```

    Credentials: client.NewAPIKeyStaticCredentials(apiKey),
}
c, err := client.Dial(clientOptions)

> if err != nil {
>   log.Fatalf("error creating temporal client: %v", err)
> }

Reviewer 2: ```suggestion
```

Can probably remove error handling

## Connect to Temporal Cloud {#connect-to-temporal-cloud}

### How to connect to Temporal Cloud using an API Key {#connect-to-temporal-cloud-api-key}

> To use an [API key](/cloud/api-keys) with the Temporal Go SDK, you will need to gather some information first:

Author: Agreed. Fixing.

- When you create your _API Key,_ be sure to save the value.
- You can see the _namespace account id_ combination in your URL or anywhere that you manage namespaces.
  It follows the format `<namespace_id>.<account_id>`.
- The _endpoint_ will vary. The most common endpoint used is the gRPC regional endpoint, which you can find by selecting the Connect button on the namespace details page.
  It follows the format: `<region>.<cloud_provider>.api.temporal.io:7233`.
- For Namespaces with High Availability features with API key authentication enabled, use the gRPC Namespace endpoint: `<namespace>.<account>.tmprl.cloud:7233`.
  This allows automated failover without needing to switch endpoints.

> Once you have the necessary info, to create an initial connection, use this code:

Reviewer 3: Would rephrase this a little bit:

```suggestion
Now, when instantiating a Temporal `client` in your Go SDK code, provide the API key with the following `ConnectionOptions`:
```

```go
clientOptions := client.Options{
    HostPort: <endpoint>,
    Namespace: <namespace_id>.<account_id>,
    ConnectionOptions: client.ConnectionOptions{TLS: &tls.Config{}},
    Credentials: client.NewAPIKeyStaticCredentials(apiKey),
}
c, err := client.Dial(clientOptions)
```

> To update an API key:

Reviewer 3: ```suggestion
If you need to update an API key, use the Go `context` object:
```

Author: Agreed; changed.

## Connect to Temporal Cloud {#connect-to-temporal-cloud}

### How to connect to Temporal Cloud using an API Key

> To use your [API key](/cloud/api-keys) with a Temporal SDK, include the API key in the Bearer token header and specify the Namespace header.
> The Bearer token header indicates the authentication type, while the Namespace header routes the request.

Reviewer: ```suggestion
To use an [API key](/cloud/api-keys) with Temporal Go SDK, leverage the following code example.
```

Create an initial connection:

```go
clientOptions := client.Options{
    HostPort: <endpoint>,
    Namespace: <namespace_id>.<account_id>,

>     ConnectionOptions: client.ConnectionOptions{
>         TLS: &tls.Config{},
>         DialOptions: []grpc.DialOption{
>             grpc.WithUnaryInterceptor(
>                 func(ctx context.Context, method string, req any, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
>                     return invoker(
>                         metadata.AppendToOutgoingContext(ctx, "temporal-namespace", <namespace_id>.<account_id>),
>                         method,
>                         req,
>                         reply,
>                         cc,
>                         opts...,
>                     )
>                 },
>             ),
>         },
>     },

Reviewer: ```suggestion
    ConnectionOptions: client.ConnectionOptions{TLS: &tls.Config{},},
```

    Credentials: client.NewAPIKeyStaticCredentials(apiKey),
}
c, err := client.Dial(clientOptions)
if err != nil {
  log.Fatalf("error creating temporal client: %v", err)
}
```

To update an API key:

```go
// Assuming client Credentials created with
var myKey string
creds := client.NewAPIKeyDynamicCredentials(
    func(context.Context) (string, error) { return myKey, nil })

> // Just update by replacing

Reviewer: ```suggestion
// Update by replacing
```
